File: demo/src/main/java/com/midas/demo/mvp/contract/RankContract.java
package com.midas.demo.mvp.contract;

import com.midas.demo.base.IPresenter;
import com.midas.demo.base.IView;
import com.midas.demo.mvp.model.bean.HomeBean;

import java.util.List;

public interface RankContract {

    interface View extends IView{

        void setRankList(List<HomeBean.Issue.Item> itemList);

        void setError(String error);
    }

    interface Presenter extends IPresenter<View>{

        void getRankList(String url);
    }
}

File: app/src/main/java/com/midas/eye/mvp/contract/RankContract.kt
package com.midas.eye.mvp.contract

import com.midas.eye.base.IBaseView
import com.midas.eye.base.IPresenter
import com.midas.eye.mvp.model.bean.HomeBean

/**
 * Created by midas on 2019/05/28.
 * desc: 契约类
 */
interface RankContract {

    interface View: IBaseView {
        /**
         * 设置排行榜的数据
         */
        fun setRankList(itemList: ArrayList<HomeBean.Issue.Item>)

        fun showError(errorMsg:String,errorCode:Int)
    }


    interface Presenter: IPresenter<View> {
        /**
         * 获取 TabInfo
         */
        fun requestRankList(apiUrl:String)
    }
}